java/service_instance.go

```diff
@@ -803,12 +809,22 @@ type CreateServiceInstanceInput struct {
 	// you can first create reserved IP addresses, then provision the service instance to use those persistent IP addresses.
 	// Optional.
 	IPNetwork string `json:"ipNetwork,omitempty"`
-	// This attribute is not available on Oracle Cloud Infrastructure.
-	// This attribute is applicable only when provisioning an Oracle Java Cloud Service instance that uses
-	// Oracle Identity Cloud Service to configure user authentication and administer users, groups, and roles.
-	// Groups properties for the Oracle managed load balancer.
-	// Optional
+	// Groups properties for an Oracle-managed load balancer, which is maintained and patched by Oracle.
+	// On Oracle Cloud Infrastructure Classic and Oracle Cloud at Customer: This attribute is only applicable if provisioning a service instance with useIdentityService set to true.
+	// When Oracle Identity Cloud Service is enabled, the Oracle-managed load balancer runs on Oracle Cloud Infrastructure Load Balancing Classic. You can set the load balancing policy.
+	// On Oracle Cloud Infrastructure: This attribute is only applicable if provisioning a service instance with configureLoadBalancer set to true,
+	// where the Oracle-managed load balancer runs on Oracle Cloud Infrastructure Load Balancing. You can set the load balancing policy.
+	// If a specific subnet is specified for the service instance, you can also specify subnets for the Oracle-managed load balancer nodes (node 1 and node 2).
+	// Optional.
 	LoadBalancer *LoadBalancer `json:"loadbalancer,omitempty"`
+	// Type of Oracle-managed load balancer. Valid values are:
+	// PRIVATE - The Oracle-managed load balancer is not accessible over the Internet. Note that this type is not available on Oracle Cloud Infrastructure.
+	// PUBLIC - The Oracle-managed load balancer is accessible over the Internet. This is the default.
+	// This attribute is only applicable when creating an Oracle Java Cloud Service instance with an Oracle-managed load balancer.
+	// On Oracle Cloud Infrastructure Classic: This attribute is only applicable when provisioning a service instance on a specific IP network (ipNetwork)
+	// and Oracle Identity Cloud Service is enabled (where useIdentityService is set to true).
+	// Optional.
+	LoadBalancerType string `json:"loadBalancerType,omitempty"`
```

Comment:
Thoughts on swapping this to using constants?

Reply:
Good catch. Added `ServiceInstanceLoadBalancerType` constant.